### ISSUE TITLE
loss suddenly increase?

### ISSUE THREAD

**Issue by jacklin602** [lilanxiao/Rotated_IoU#32]
Hi @lilanxiao,
thanks for your great work!
When I run the demo, I found that sometimes loss will suddenly increase and iou will suddenly drop.
Is there any ideas about this problem?
![differentiableIOU_loss_problem](https://user-images.githubusercontent.com/49774448/134863730-8ac1c752-b477-448a-87c4-fff90c3d3c5b.png)

**Comment by lilanxiao** [comment]
hi, thank you for this issue! 
Yes, it also happens to me sometimes. I think it's due to my non-optimal network design and hyper-parameter choices (e.g. too large learning rate), which leads to unstable training. I made this demo to show that the IoU loss is differentiable and didn't pay much attention to details. 
But I don't think it's a big problem. I've used the IoU loss to train some 3D detectors. It worked well and didn't show any instability.
